tests/testLib.cc

```diff
@@ -0,0 +1,59 @@
+#include <string>
+
+#include <pybind11/pybind11.h>
+
+#include "lsst/pex/exceptions/Exception.h"
+#include "lsst/pex/exceptions/Runtime.h"
+#include "lsst/pex/exceptions/python/Exception.h"
+
+using namespace lsst::pex::exceptions;
+
+LSST_EXCEPTION_TYPE(TestError, lsst::pex::exceptions::RuntimeError, TestError)
+
+template <typename T>
+void fail1(std::string const & message) {
+    throw LSST_EXCEPT(T, message);
+}
+
+template <typename T>
+void fail2(std::string const & message1, std::string const & message2) {
+    try {
+        fail1<T>(message1);
+    } catch (T & err) {
+        LSST_EXCEPT_ADD(err, message2);
+        throw err;
+    }   
+}
+
+#define LSST_FAIL_TEST(name) \
+    mod.def("fail" #name "1", [](const std::string & message){ \
+            fail1<name>(message); \
+    }); \
+    mod.def("fail" #name "2", [](const std::string & message1, const std::string & message2){ \
+            fail2<name>(message1, message2); \
+    });
```

Comment:
So this is annoying. Because `name` is used both as the type and to construct the function name. Although this can be done with traits I think it is easier just to stick with the macro. It also is only used in this one test library file so I don't see the harm.